src/components/Textarea.react.js

```diff
@@ -28,26 +28,37 @@ export default class Textarea extends Component {
                     if (setProps) {
                         setProps({value: e.target.value});
                     }
-                    if (fireEvent) {
-                        fireEvent({event: 'change'});
-                    }
                 }}
                 onBlur={() => {
-                    if (fireEvent) {
-                        fireEvent({event: 'blur'});
+                    if (setProps) {
+                        setProps({
+                            n_blur: this.props.n_blur + 1,
+                            n_blur_timestamp: new Date(),
+                        });
                     }
                 }}
                 onClick={() => {
-                    if (fireEvent) {
-                        fireEvent({event: 'click'});
+                    if (setProps) {
+                        setProps({
+                            n_clicks: this.props.n_clicks + 1,
+                            n_clicks_timestamp: new Date(),
+                        });
                     }
                 }}
-                {...omit(['fireEvent', 'setProps', 'value'], this.props)}
+                {...omit(['setProps', 'value'], this.props)}
             />
         );
     }
 }
 
+Textarea.defaultProps = {
+    n_blur: 0,
+    n_blur_timestamp: -1,
+    n_clicks: 0,
+    n_clicks_timestamp: -1,
```

Comment:
:smiley_cat: This addition of props should be noted somewhere, had to read the code to see it.

Reply:
Thanks @T4rk1n -> added to `CHANGELOG` in https://github.com/plotly/dash-core-components/pull/444/commits/8b4936b62534bf4fc539c269a070c824d3337dc3

Kind of amazing that these two are the only events we had that weren't covered by an existing regular prop already.